frontend/lib/src/components/graph.dart

```diff
@@ -19,138 +18,213 @@ class Graph extends StatelessWidget {
+                                          pieTouchResponse.touchedSection ==
+                                              null) {
+                                        graphController.setFocusIndex = -1;
+
+                                        return;
+                                      }
+                                      graphController.setFocusIndex =
+                                          pieTouchResponse.touchedSection!
+                                              .touchedSectionIndex;
+                                    });
+                                  },
+                                ),
+                                borderData: FlBorderData(
+                                  show: false,
+                                ),
+                                sectionsSpace: 0,
+                                centerSpaceRadius: 30,
+                                sections: showingSections(),
+                              ),
+                            ),
+                          );
+                        },
+                      ),
                     ),
-                    const SizedBox(
-                      height: 18,
+                    Column(
+                      mainAxisAlignment: MainAxisAlignment.end,
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: legendSection(legendList),
                     ),
                   ],
                 ),
-              ],
+              ),
             ),
           ),
-        ),
-      ),
+        );
+      },
     );
   }
 
+  List<Widget> legendSection(List<TransactionCategory> legendList) {
+    List<Widget> legends = [];
+    if (legendList.contains(TransactionCategory.FOOD)) {
+      legends.add(
+        Indicator(
+          color: const Color(0xFF0384EA),
+          text: TransactionCategory.FOOD.value,
+          isSquare: true,
+        ),
+      );
+    }
+    if (legendList.contains(TransactionCategory.TRANSPORTATION)) {
+      legends.add(
+        Indicator(
+          color: const Color(0xFFFEA42C),
+          text: TransactionCategory.TRANSPORTATION.value,
+          isSquare: true,
+        ),
+      );
+    }
+    if (legendList.contains(TransactionCategory.BILLS)) {
+      legends.add(
+        Indicator(
+          color: const Color(0xFF8047F6),
+          text: TransactionCategory.BILLS.value,
+          isSquare: true,
+        ),
+      );
+    }
+    if (legendList.contains(TransactionCategory.SAVINGS)) {
+      legends.add(
+        Indicator(
+          color: const Color(0xFF00D27C),
+          text: TransactionCategory.SAVINGS.value,
+          isSquare: true,
+        ),
+      );
+    }
+    if (legendList.contains(TransactionCategory.MISC)) {
+      legends.add(
+        Indicator(
+          color: const Color(0xFFDC4949),
+          text: TransactionCategory.MISC.value,
+          isSquare: true,
+        ),
+      );
```

Comment:
```dart
  List<Widget> legendSection(List<SpendingBreakdown> spendingList) {
    List<Widget> legends = [];

    const transactionTypeToColor = {
      TransactionCategory.FOOD: Color(0xFF0384EA),
      TransactionCategory.TRANSPORTATION: Color(0xFFFEA42C),
      TransactionCategory.BILLS: Color(0xFF8047F6),
      TransactionCategory.SAVINGS: Color(0xFF00D27C),
      TransactionCategory.MISC: Color(0xFFDC4949),
    };

    for (var breakdown in spendingList) {
      final totalSpentPercentage = ((breakdown.totalTransactionAmount /
              spendingBreakdownController.totalSpent) *
          100);
      final percentageString = totalSpentPercentage.toStringAsFixed(1);

      final color =
          transactionTypeToColor[breakdown.category] ?? const Color(0xFFDC4949);
      final text = (totalSpentPercentage < 5)
          ? '${breakdown.category.value} ($percentageString%)'
          : breakdown.category.value;

      legends.add(
        Indicator(color: color, text: text, isSquare: true),
      );
    }
```

i wrote this up, its much cleaner and less repetetive than legendList.contains implementation for each transactioncategory